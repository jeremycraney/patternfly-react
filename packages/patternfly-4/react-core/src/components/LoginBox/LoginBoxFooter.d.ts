import { SFC, HTMLProps, ReactNode } from 'react';

export interface LoginBoxFooterProps extends HTMLProps<HTMLDivElement> {
  children?: ReactNode;
  className?: string;
}

declare const LoginBoxFooter: SFC<LoginBoxFooterProps>;

export default LoginBoxFooter;
