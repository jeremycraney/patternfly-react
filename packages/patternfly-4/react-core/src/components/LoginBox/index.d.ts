export { default as LoginBox, LoginBoxProps } from './LoginBox';
export { default as LoginBoxHeader, LoginBoxHeaderProps } from './LoginBoxHeader';
export { default as LoginBoxBody, LoginBoxBodyProps } from './LoginBoxBody';
export { default as LoginBoxFooter, LoginBoxFooterProps } from './LoginBoxFooter';
