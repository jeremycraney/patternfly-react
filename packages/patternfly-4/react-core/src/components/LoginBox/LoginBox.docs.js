import { LoginBox, LoginBoxHeader, LoginBoxBody, LoginBoxFooter } from '@patternfly/react-core';
import Simple from './examples/SimpleLoginBox';

export default {
  title: 'LoginBox',
  components: {
    LoginBox,
    LoginBoxHeader,
    LoginBoxBody,
    LoginBoxFooter
  },
  examples: [Simple]
};
