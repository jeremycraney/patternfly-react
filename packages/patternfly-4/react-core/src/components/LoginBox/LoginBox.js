import React from 'react';
import styles from '@patternfly/patternfly-next/components/LoginBox/login-box.css';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';

const propTypes = {
  /** Content rendered inside the LoginBox */
  children: PropTypes.node,
  /** Additional classes added to the LoginBox */
  className: PropTypes.string
};

const defaultProps = {
  children: null,
  className: ''
};

const LoginBox = ({ className, children, ...props }) => (
  <div {...props} className={css(styles.loginBox, className)}>
    {children}
  </div>
);

LoginBox.propTypes = propTypes;
LoginBox.defaultProps = defaultProps;

export default LoginBox;
