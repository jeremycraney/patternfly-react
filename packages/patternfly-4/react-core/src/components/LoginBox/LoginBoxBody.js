import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/patternfly-next/components/LoginBox/login-box.css';

const propTypes = {
  /** Content rendered inside the LoginBox Body */
  children: PropTypes.node,
  /** Additional classes added to the LoginBox Body */
  className: PropTypes.string
};

const defaultProps = {
  children: null,
  className: ''
};

const LoginBoxBody = ({ children, className, ...props }) => (
  <div className={css(styles.loginBoxBody, className)} {...props}>
    {children}
  </div>
);

LoginBoxBody.propTypes = propTypes;
LoginBoxBody.defaultProps = defaultProps;

export default LoginBoxBody;
