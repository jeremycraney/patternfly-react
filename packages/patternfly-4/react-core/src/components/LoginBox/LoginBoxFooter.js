import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/patternfly-next/components/LoginBox/login-box.css';

const propTypes = {
  /** Content rendered inside the LoginBox Footer */
  children: PropTypes.node,
  /** Additional classes added to the LoginBox Footer */
  className: PropTypes.string
};

const defaultProps = {
  children: null,
  className: ''
};

const LoginBoxFooter = ({ children, className, ...props }) => (
  <div className={css(styles.loginBoxFooter, className)} {...props}>
    {children}
  </div>
);

LoginBoxFooter.propTypes = propTypes;
LoginBoxFooter.defaultProps = defaultProps;

export default LoginBoxFooter;
