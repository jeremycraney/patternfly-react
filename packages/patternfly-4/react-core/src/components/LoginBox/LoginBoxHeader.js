import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/patternfly-next/components/LoginBox/login-box.css';

const propTypes = {
  /** Content rendered inside the LoginBox Header */
  children: PropTypes.node,
  /** Additional classes added to the LoginBox Header */
  className: PropTypes.string
};

const defaultProps = {
  children: null,
  className: ''
};

const LoginBoxHeader = ({ children, className, ...props }) => (
  <div className={css(styles.loginBoxHeader, className)} {...props}>
    {children}
  </div>
);

LoginBoxHeader.propTypes = propTypes;
LoginBoxHeader.defaultProps = defaultProps;

export default LoginBoxHeader;
