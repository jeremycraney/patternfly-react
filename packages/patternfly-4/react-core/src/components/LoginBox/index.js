export { default as LoginBox } from './LoginBox';
export { default as LoginBoxHeader } from './LoginBoxHeader';
export { default as LoginBoxBody } from './LoginBoxBody';
export { default as LoginBoxFooter } from './LoginBoxFooter';
