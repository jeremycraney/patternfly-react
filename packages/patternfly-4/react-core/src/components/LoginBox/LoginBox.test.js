import React from 'react';
import { shallow, mount } from 'enzyme';
import LoginBox from './LoginBox';
import LoginBoxBody from './LoginBoxBody';
import LoginBoxHeader from './LoginBoxHeader';
import LoginBoxFooter from './LoginBoxFooter';

const LoginBoxContent = () => (
  <React.Fragment>
    <LoginBox>
      <LoginBoxHeader>LoginBox Header</LoginBoxHeader>
      <LoginBoxBody>LoginBox Body</LoginBoxBody>
      <LoginBoxFooter>LoginBox Footer</LoginBoxFooter>
    </LoginBox>
  </React.Fragment>
);

test('simple LoginBox', () => {
  const view = mount(<LoginBoxContent />);
  expect(view).toMatchSnapshot();
});

test('renders with PatternFly Core styles', () => {
  const view = shallow(<LoginBox />);
  expect(view).toMatchSnapshot();
});

test('className is added to the root element', () => {
  const view = shallow(<LoginBox className="extra-class" />);
  expect(view.prop('className')).toMatchSnapshot();
});

test('extra props are spread to the root element', () => {
  const testId = 'loginbox';
  const view = shallow(<LoginBox data-testid={testId} />);
  expect(view.prop('data-testid')).toBe(testId);
});
