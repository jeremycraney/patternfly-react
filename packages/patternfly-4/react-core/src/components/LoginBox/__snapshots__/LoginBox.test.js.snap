// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`className is added to the root element 1`] = `"pf-c-login-box extra-class"`;

exports[`renders with PatternFly Core styles 1`] = `
.pf-c-login-box {
  display: flex;
  flex-direction: column;
  background-color: #ffffff;
}

<div
  className="pf-c-login-box"
/>
`;

exports[`simple LoginBox 1`] = `
.pf-c-login-box__header {
  display: block;
  padding: 3rem 3rem 1rem 3rem;
}
.pf-c-login-box__body {
  display: block;
  padding: 2rem 3rem 2rem 3rem;
}
.pf-c-login-box__footer {
  display: block;
  padding: 2rem 3rem 4rem 3rem;
}
.pf-c-login-box {
  display: flex;
  flex-direction: column;
  background-color: #ffffff;
}

<LoginBoxContent>
  <LoginBox
    className=""
  >
    <div
      className="pf-c-login-box"
    >
      <LoginBoxHeader
        className=""
      >
        <div
          className="pf-c-login-box__header"
        >
          LoginBox Header
        </div>
      </LoginBoxHeader>
      <LoginBoxBody
        className=""
      >
        <div
          className="pf-c-login-box__body"
        >
          LoginBox Body
        </div>
      </LoginBoxBody>
      <LoginBoxFooter
        className=""
      >
        <div
          className="pf-c-login-box__footer"
        >
          LoginBox Footer
        </div>
      </LoginBoxFooter>
    </div>
  </LoginBox>
</LoginBoxContent>
`;
